Sources/PrivoSDK/api/Rest.swift

```diff
@@ -148,22 +172,36 @@ class Rest {
     }
     
     func trackPossibleAFError(_ error: AFError?, _ response: String?, _ code: Int?) {
+        let encoder = JSONEncoder()
+        var analyticErrorEvent: AnalyticEventErrorData? = nil
+        
         if (code != 200 && code != 204 && code != 205) {
-            if let error = error {
-                let data = AnalyticEventErrorData(errorMessage: error.errorDescription, response: response, errorCode: error.responseCode, privoSettings: nil)
-                if let jsonData = try? JSONEncoder().encode(data) {
-                    let jsonString = String(decoding: jsonData, as: UTF8.self)
-                    let event = AnalyticEvent(serviceIdentifier: PrivoInternal.settings.serviceIdentifier, data: jsonString)
-                    sendAnalyticEvent(event)
-                }
-            }
+            // This branch for HTTP errors.
+            analyticErrorEvent = AnalyticEventErrorData(errorMessage: nil,
+                                                        response: response,
+                                                        errorCode: code,
+                                                        privoSettings: nil)
+        } else if let error = error {
+            // This branch for:
+            // - underlying errors like network request timeout,
+            // - (overlying) decoding parsing error.
```

Comment:
like comments. Maybe just split this logic to 2 separate functions ?

Reply:
To honestly divide this function into two parts of error handling, it is necessary to take out the error checking of the lower level (network communication: availability, timeouts, cancelled requests, ... etc) and perform it before parsing. Right now, I feel this will require deep intervention and refactoring. Let's keep this issue in mind. Usually, the desire to solve this kind of problem accumulates and will still make itself felt.